packages/types/src/proving-keys.ts

```diff
@@ -1,9 +1,32 @@
-export const provingKeys = [
-  { keyType: 'spend', file: 'spend_pk.bin' },
-  { keyType: 'output', file: 'output_pk.bin' },
-  { keyType: 'swap', file: 'swap_pk.bin' },
-  { keyType: 'swap_claim', file: 'swapclaim_pk.bin' },
-  { keyType: 'nullifier_derivation', file: 'nullifier_derivation_pk.bin' },
-  { keyType: 'delegator_vote', file: 'delegator_vote_pk.bin' },
-  { keyType: 'convert', file: 'convert_pk.bin' },
-];
+import { Action } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/transaction/v1/transaction_pb';
+
+type ActionType = Exclude<Action['action']['case'], undefined>;
+export interface ProvingKey {
+  keyType: string;
+  file: string;
+}
+
+export const provingKeysByActionType: Record<ActionType, ProvingKey | null> = {
```

Comment:
Can we convert the `null`'s to `undefined`? Not sure how you feel, but I lean toward the opinion that null was a mistake in the javascript language.

Reply:
Another option is to make this record a partial so it doesn't need them all enumerated. But perhaps it's better that we see them all? 🤷

Reply:
Ah, good call — I made them undefined. I like the idea that, even for us coders' sake, we see all the action names and what keys are required for each, even if the answer is "none."